### ISSUE TITLE
Go vendoring

### ISSUE THREAD

**Issue by ilovezfs** [getgauge/gauge#729]
The Homebrew formula for gauge will apparently now need 20 resource blocks in order to be upgraded to 0.9.0.

Previously we needed zero, since the formula would simply run `godep restore`.

Updating 20+ resource blocks each new release will not really be sustainable in terms of maintenance. In the absence of Go vendoring, this is the only way we can ensure reproducible builds.

Would you be willing to start using the Go vendor directory or one of the other Go vendor systems so that we do not need to start using resource blocks? Thanks!

See https://github.com/Homebrew/homebrew-core/pull/15835.

**Comment by sriv** [comment]
Gauge used to have `godep`. We removed it in favour of #698, allowing gauge to be built via `go get`. 

@kashishm can we piggyback on this build mechanism?

**Comment by kashishm** [comment]
@sriv Running `go get` in the Homebrew formula works, but `brew audit --strict gauge` shows that we should use `go_resource` instead of `go get`.
```
gauge:
  * C: 38: col 7: Formulae should not use `go get`. If non-vendored resources are required use `go_resource`s.
Error: 1 problem in 1 formula
```

**Comment by ilovezfs** [comment]
@sriv What @kashishm said is correct. `go get` is not allowed in Homebrew at all because it builds from tip of tree of revisions, and so does not lead to reproducible or secure builds.

**Comment by ilovezfs** [comment]
If you found `godep` problematic for some reason, you might look at

https://github.com/Masterminds/glide (`glide install`)
or
https://github.com/golang/dep (`dep ensure`)

as alternatives.

**Comment by ilovezfs** [comment]
There is also

**Comment by ilovezfs** [comment]
There is also https://github.com/kardianos/govendor (`govendor sync`)

**Comment by sriv** [comment]
Thanks for the suggestions and feedback @ilovezfs. We have re-introduced `godep`. You raised a valid point about reproducible builds. 

This is fixed via #731. We will raise a new PR for the homebrew formula for gauge. Thanks.

**Comment by ilovezfs** [comment]
@sriv thank you! :heart:
